### PR TITLE
DOC: ecosystem pandas-stubs information

### LINKED ISSUE #43197: How to make pandas-stubs library compliant with pandas development team requirements
Hello,

my name is Zbigniew Królikowski. I am a software developer at VirtusLab.

Together with my colleagues we started working on our own [pandas type stubs library](https://github.com/VirtusLab/pandas-stubs), and at some point decided to make it open-source.

We wanted to ask you:
1. Is the separation between our library and pandas itself clear? We wanted to be clear on the authorship so we added a [disclaimer](https://github.com/VirtusLab/pandas-stubs).
2. Do you see any issues with how we promote our library, especially in terms of our graphics?

We'd be very grateful to hear your thoughts on this.

Best regards,
Zbigniew Królikowski

### PR DESCRIPTION
**Pull request by zkrolikowski-vl**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Added information about pandas-stubs to both the .rst and .md ecosystem doc files.

closes https://github.com/pandas-dev/pandas/issues/43197

### CODE REVIEW

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -575,3 +575,20 @@ Library            Accessor     Classes                              Description
...
+
+
+Type stubs
+----------------------------
+
+`pandas-stubs <https://github.com/VirtusLab/pandas-stubs>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pandas by itself doesn't expose any type information to the user.
+This means that in pandas-dependant projects the type checkers like Mypy won't perform like expected
```
Comment (joannasendorek) [inline doc/source/ecosystem.rst:587]: dependent

Comment (joannasendorek) [inline doc/source/ecosystem.rst:587]:
```suggestion
This means that in pandas-dependent projects the type checkers like Mypy won't perform like expected
```

Comment (joannasendorek) [inline doc/source/ecosystem.rst:587]:
```suggestion
This means that in pandas-dependant projects the type checkers like mypy won't perform like expected
```

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -575,3 +575,20 @@ Library            Accessor     Classes                              Description
...
+and will treat all pandas objects as they were of type ``Any``.
+Learn more by reading through these issues `14468 <https://github.com/pandas-dev/pandas/issues/14468>`_,
+`26766 <https://github.com/pandas-dev/pandas/issues/26766>`_, `28142 <https://github.com/pandas-dev/pandas/issues/28142>`_.
+
+pandas-stubs are meant to alleviate this issue by providing rudimentary type coverage of pandas API.
```
Comment (joannasendorek) [inline doc/source/ecosystem.rst:592]: basic?

**Review by joannasendorek — APPROVED** [review]

**Review by datapythonista — COMMENTED** [review]
Looks good, couple of ideas:

Maybe instead of a section `Type stubs` that it's likely to never have anything more than this project, a section development tools (or a similar name) could be used. I know there were some interest in a package for pandas stuff for users of certain IDEs. I think having those together could make sense.

Then, I wonder if having a usage example would help more than this technical information. It's clear to me, but I wonder if many of our users are not familiar with type annotations, and it won't be immediately obvious what this means. It's a personal opinion, and you can surely disagree with, but I feel like providing a short sentence saying that pandas-stubs enables type checking in pandas should be enough. If then we just have an example of installing with pip/conda, a sample file using pandas with annotations, and a call to mypy (maybe a call with and without installing pandas-stubs.

In any case, happy to get it merged like this too.

**Review by datapythonista — APPROVED** [review]
Great example, very clear. Thanks for the update.

**Comment by zkrolikowski-vl** [comment]
@datapythonista Do I have to take some action for the CI to pass. It times-out after an hour.

**Comment by alimcmaster1** [comment]
> @datapythonista Do I have to take some action for the CI to pass? It times-out after an hour.

I have just restarted CI now. Hopefully should run to completion this time

**Comment by phofl** [comment]
pre-commit raises errors. You should fix them first

File: doc/source/ecosystem.rst [hunk 3]
```diff
@@ -575,3 +575,75 @@ Library            Accessor     Classes                              Description
...
+----------------------------
+
+`pandas-stubs <https://github.com/VirtusLab/pandas-stubs>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pandas doesn't expose any type information to the user by itself.
+Install pandas-stubs to enable basic type coverage of pandas API.
+
+Learn more by reading through these issues `14468 <https://github.com/pandas-dev/pandas/issues/14468>`_,
+`26766 <https://github.com/pandas-dev/pandas/issues/26766>`_, `28142 <https://github.com/pandas-dev/pandas/issues/28142>`_.
```
Comment (jreback) [inline doc/source/ecosystem.rst:590]: can you remove from here on down

Comment (jreback) [reply doc/source/ecosystem.rst:590]: you can just refer to the github page / docs

File: doc/source/ecosystem.rst [hunk 4]
```diff
@@ -575,3 +575,75 @@ Library            Accessor     Classes                              Description
...
 .. _woodwork: https://github.com/alteryx/woodwork
+
+
+Development tools
+----------------------------
+
+`pandas-stubs <https://github.com/VirtusLab/pandas-stubs>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pandas doesn't expose any type information to the user by itself.
```
Comment (jreback) [inline doc/source/ecosystem.rst:586]: this is not true there is lots of type information, explain how this is different.

File: web/pandas/community/ecosystem.md [hunk 5]
```diff
@@ -388,3 +388,65 @@ authors to coordinate on the namespace.
   | [pint-pandas](https://github.com/hgrecco/pint-pandas)                | `pint`     | `Series`, `DataFrame` |
   | [composeml](https://github.com/alteryx/compose)                      | `slice`    | `DataFrame`           |
   | [woodwork](https://github.com/alteryx/woodwork)                      | `slice`    | `Series`, `DataFrame` |
+
```
Comment (jreback) [inline web/pandas/community/ecosystem.md:391]: umm what is this?

**Review by jreback — CHANGES_REQUESTED** [review]

File: web/pandas/community/ecosystem.md (hunk 5 above)
Comment (zkrolikowski-vl) [reply web/pandas/community/ecosystem.md]: If you mean the empty line I removed it. I'm using Unix line endings.

File: doc/source/ecosystem.rst (hunk 3 above)
Comment (zkrolikowski-vl) [reply doc/source/ecosystem.rst:589]: was originally like this but extended it on advice of an other maintainer. Went back.

File: doc/source/ecosystem.rst (hunk 4 above)
Comment (zkrolikowski-vl) [reply doc/source/ecosystem.rst]: is this better? `While pandas repository is partially typed, the package itself doesn't expose this information for external use. `

**Comment by zkrolikowski-vl** [comment]
I will get around to fixing the pre-commit issues locall.y

**Review by lithomas1 — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @zkrolikowski-vl

**Pull request merged**